account_product_fiscal_classification/model/account_product_fiscal_classification.py

```diff
@@ -0,0 +1,162 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Account Product - Fiscal Classification module for Odoo
+#    Copyright (C) 2014 -Today GRAP (http://www.grap.coop)
+#    @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+from openerp import models, fields, api, _
+from openerp.exceptions import ValidationError
+
+
+class AccountProductFiscalClassification(models.Model):
+    """Fiscal Classification of customer and supplier taxes.
+    This classification is linked to a product to select a bundle of taxes
+     in one time."""
+    _name = 'account.product.fiscal.classification'
+    _description = 'Product Fiscal Classification'
+    _MAX_LENGTH_NAME = 256
+
+    # Default Section
+    def _default_company_id(self):
+        return self.env['res.users']._get_company()
+
+    # Field Section
+    name = fields.Char(
+        size=_MAX_LENGTH_NAME, required=True, select=True)
+
+    company_id = fields.Many2one(
+        comodel_name='res.company', default=_default_company_id,
+        string='Company', help="Specify a company"
+        " if you want to define this Fiscal Classification only for specific"
+        " company. Otherwise, this Fiscal Classification will be available"
+        " for all companies.")
+
+    active = fields.Boolean(
+        default=True,
+        help="If unchecked, it will allow you to hide the Fiscal"
+        " Classification without removing it.")
+
+    product_tmpl_ids = fields.One2many(
+        comodel_name='product.template', string='Products',
+        compute='_compute_product_tmpl_info')
+
+    product_tmpl_qty = fields.Integer(
+        string='Products Quantity', compute='_compute_product_tmpl_info')
+
+    purchase_tax_ids = fields.Many2many(
```

Comment:
FYI : about https://github.com/OCA/account-fiscal-rule/pull/24#discussion_r38878743
I set purchase_tax_ids instead of purchase_base_tax_ids, for clarity. I used oldname and I set the same table name, so upgrade should work.
If your custom modules overload this field and sale_base_tax_ids, please make the changes.